02_bubble/nicholascross/main.go

```diff
@@ -0,0 +1,52 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"os"
+)
+
+var out io.Writer = os.Stdout
+
+func main() {
+	fmt.Fprintln(out, bubble([]int{3, 2, 1, 5}))
+	fmt.Fprintln(out, insertion([]int{3, 2, 1, 5}))
+}
+
+func bubble(s []int) []int {
+	count := len(s)
+	sort := make([]int, count)
+	copy(sort, s)
+
+	for {
+		swapped := false
+
+		for i := 0; i+1 < count; i++ {
+			if sort[i] > sort[i+1] {
+				sort[i+1], sort[i] = sort[i], sort[i+1]
+				swapped = true
+			}
+		}
+
+		if !swapped {
```

Comment:
Just an observation - this case could be moved to the `for` clause so it does not initially present as an infinite loop:
```
swapped := true
for swapped {
  swapped = false
  ...
}
```

That does not read quite as well, so a different variable name may make sense:
```
sorted := false
for !sorted {
  sorted = true // assume we're sorted unless we find an element to swap
  ...
    sorted = false
}
```

I am not requesting a change here, just commenting.